flexmeasures/data/services/scheduling.py

```diff
@@ -204,6 +206,61 @@ def make_schedule(
     return True
 
 
+def load_custom_scheduler(scheduler_specs: dict) -> Tuple[Callable, str]:
+    """
+    Read in custom scheduling spec.
+    Attempt to load the Callable, also derive a data source name.
+
+    Example specs:
+
+    {
+        "module": "/path/to/module.py",  # or sthg importable, e.g. "package.module"
+        "function": "name_of_function",
+        "source": "source name"
+    }
+
+    """
+    assert isinstance(
+        scheduler_specs, dict
+    ), f"Scheduler specs is {type(scheduler_specs)}, should be a dict"
+    assert "module" in scheduler_specs, "scheduler specs have no 'module'."
+    assert "function" in scheduler_specs, "scheduler specs have no 'function'"
+
+    source_name = scheduler_specs.get(
+        "source", f"Custom scheduler - {scheduler_specs['function']}"
+    )
+    scheduler_name = scheduler_specs["function"]
+
+    # import module
+    module_descr = scheduler_specs["module"]
+    if os.path.exists(module_descr):
+        spec = importlib.util.spec_from_file_location(scheduler_name, module_descr)
+        assert spec, f"Could not load specs for scheduling module at {module_descr}."
+        module = importlib.util.module_from_spec(spec)
+        sys.modules[scheduler_name] = module
+        assert isinstance(spec.loader, Loader)
+        spec.loader.exec_module(module)
+    else:  # assume importable module
+        try:
+            module = importlib.import_module(module_descr)
+        except TypeError:
+            current_app.log.error(f"Cannot load {module_descr}.")
+            raise
+        except ModuleNotFoundError:
+            current_app.logger.error(
+                f"Attempted to import module {module_descr} (as it is not a valid file path), but it is not installed."
+            )
+            raise
+    assert module, f"Module {module_descr} could not be loaded."
```

Comment:
This seems to make lines 246-246 obsolete, given that the extra error logging in those lines doesn't seem to be more explicit than the assertion error that would otherwise be raised here.